### PR TITLE
OOP school library: add Person, Student and Teacher classes.

### PR DESCRIPTION
**Pull request by Irirwanirira**
## This Pr consists of: 

- [X] I created a person class with its parameters
- [X] I set atrr_reader and atrr_writer
- [x] I created a class of teachers inherits from a person in its file and gave its conditions 
- [x] I created a student class in its file inherited from a person class and gave hold to its parameter.

### Thank you for reviewing my project.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(name, age, classroom, parent_permission)
+    super(name, age, parent_permission)
+    @classroom = classroom
+  end
```
Comment (mikethreels) [inline student.rb]: - [x] the default values for name and parent_permission should be included here as well.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(name, age, specialization, parent_permission)
+    super(name, age, parent_permission)
+    @specialization = specialization
+  end
```
Comment (mikethreels) [inline teacher.rb]: - [ ] the default values for name and parent_permission should be included here as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Joseph, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 2 above)
Comment (Irirwanirira) [reply teacher.rb]: @mikethreels Thank you!

File: student.rb (hunk 1 above)
Comment (Irirwanirira) [reply student.rb]: Thank you !

**Review by prolajumokeoni — APPROVED** [review]
Hi @Irirwanirira 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**